compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -2529,12 +2533,32 @@ LogicalResult initGPULaunchConfig(FunctionOpInterface funcOp) {
         break;
       }
     }
+
+    if (auto scatterOp = dyn_cast<IREE::LinalgExt::ScatterOp>(op)) {
+      auto indices = scatterOp.getIndices();
+      if (indices.getDefiningOp() == nullptr) {
+        continue;
+      }
+      // If scatter's indices is a generic op, mark it as to skip.
+      if (auto genericOp =
+              dyn_cast<linalg::GenericOp>(indices.getDefiningOp())) {
+        genericToSkip.insert(genericOp);
+      }
+      // If scatter's backward slices are generic ops, mark them as to skip too.
+      SetVector<Operation *> slices;
+      getBackwardSlice(indices, &slices);
+      for (auto slice : slices) {
+        if (isa<linalg::GenericOp>(slice)) {
+          genericToSkip.insert(slice);
```

Comment:
We don't have to filter out generics, we can just add the `slice` directly to `genericToSkip` and make it a set of all operations used in defining the indices of the scatter.

Reply:
Sounds reasonable